devlib/utils/android.py

```diff
@@ -301,10 +301,17 @@ def __init__(self, device=None, timeout=None, platform=None, adb_server=None,
                             'poll_period': transfer_poll_period,
                             }
         self.transfer_mgr = PopenTransferManager(self, **transfer_opts) if poll_transfers else None
+        AdbConnection.active_connections[self.device] += 1
         if self.adb_as_root:
-            self.adb_root(enable=True)
+            try:
+                self.adb_root(enable=True)
+            # Exception will be raised if we are not the only connection
+            # active. adb_root() requires restarting the server, which is not
+            # acceptable if other connections are active and can apparently
+            # lead to commands hanging forever in some situations.
+            except RuntimeError:
+                pass
```

Comment:
Would it make the intention clearer here to only try and enable adb_root if the connection is not greater than 1? This would prevent hiding any other runtime errors that may occur?

Reply:
I did not want to duplicate the checking logic as adb_root() is also exposed to the user, in order to use this path on a typical run (and therefore avoid bitrot). Maybe I should make a more specific exception type though.